### PR TITLE
DM-34986: build with pyproject.toml

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,32 +1,19 @@
...
-    package_data={"": ["*.rst", "*.yaml"]},
-    scripts=["bin/run_FiberSpectrograph.py"],
-    tests_require=tests_require,
-    extras_require={"dev": dev_requires},
-    license="GPL",
-    project_urls={
-        "Bug Tracker": "https://jira.lsstcorp.org/secure/Dashboard.jspa",
-        "Source Code": "https://github.com/lsst-ts/ts_FiberSpectrograph",
-    },
+    scripts=[str(path) for path in pathlib.Path("bin").glob("*.py")],
```
Comment (timj) [inline setup.py:18]: Is the plan to switch to using entry points instead of `bin/` scripts?

Comment (r-owen) [reply setup.py:18]: I think the plan is to do that after cycle 26. In any case I suggest discussing this with Wouter and Tiago (I am happy to participate, but for now I'm just doing what I was told).

Comment (wvreeven) [reply setup.py:18]: It is a possibility that TSSW can consider doing now already instead of after cycle 26. Thanks for the suggestion, @timj!

**Review by wvreeven — APPROVED** [review]
Looks good to me.

File: bin/run_FiberSpectrograph [hunk 2]
```diff
@@ -21,8 +21,7 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 """The runner for the Fiber Spectrograph CSC."""
-import asyncio
+from lsst.ts.FiberSpectrograph.cli import execute_csc
```
Comment (timj) [inline bin/run_FiberSpectrograph:24]: Is this file kept around for testing purposes (since the installed version will come from the entry point)?

Comment (wvreeven) [reply bin/run_FiberSpectrograph:24]: That's correct.

File: setup.py (hunk 1 above)
Comment (r-owen) [reply setup.py]: Done for cycle 26, thanks to a lot of work on Wouter's part, and some on mine as well.

File: bin/run_FiberSpectrograph (hunk 2 above)
Comment (r-owen) [reply bin/run_FiberSpectrograph]: Yes, it is the only way I know of to run locally (without an installed package). If there is a way to use entry points for this, I would love to know about it.

Comment (timj) [reply bin/run_FiberSpectrograph]: `pip install -e .` does work (assuming the dependencies are available) to do a developer install that points to the working directory and I do use that for non-scons projects. The only downside is that to get it to work python installs an egg into the conda environment that then points back to your build tree. This is fine for local development but remembering to cleanup the egg (maybe `pip uninstall` will do it) is important if you don't want to be confused in the future.

Comment (timj) [reply bin/run_FiberSpectrograph]: Yes, that's what EUPS could easily set up for you. It still requires the stub file in the `bin` dir though.

**Pull request merged**